### ISSUE TITLE
Many git commands fail after deletions

### ISSUE THREAD

**Issue by The-Penultimate-Defenestrator** [ywangd/stash#159]
If I delete files locally, but don't call `git rm <file>` for each one, many many commands then fail. Without manually doing a `git rm <file>` for each one,  I get errors with `checkout`, `status`, and more. In standard git, removing files is automatically reflected.

**Comment by jsbain** [comment]
yes.  I need to restructure some things to fix this, and probably review how git behaves.  For instance, if you delete a file, it does not automatically get deleted from the index (there are some special switches that perhaps do this), but i agree we should be getting errors.    
as a temporary measure, `touch file` will at least get you past the error.

**Comment by The-Penultimate-Defenestrator** [comment]
Thanks. This works. Look forward to a fix.

**Comment by jsbain** [comment]
This was a problem in dulwich.  We rely on dulwich for a lot.  So there is sort of this trade of whether to try to fix dulwich, go the pull request route, etc, or try to have workarounds.

**Comment by The-Penultimate-Defenestrator** [comment]
Right. Makes sense.

**Comment by khilnani** [comment]
Noticed git.py in stash used a fork of dulwich. Just checking to see if the root issue was fixed in the dulwich master. Thanks!
